### PR TITLE
[REVIEW] 2D Sparse Slicing + Row & Column Indexing

### PR DESCRIPTION
**Pull request by cjnolet**
This PR builds upon #3658, adding sparse column indexing.

### CODE REVIEW

**Comment by cjnolet** [comment]
@asi1024 Just merged master into this branch. This should be ready for review.

File: cupyx/scipy/sparse/_index.py [hunk 1]
```diff
@@ -23,6 +23,155 @@
...
+    col_counts = cupy.zeros(idx_map.size, dtype=col_idxs.dtype)
+    cupyx.scatter_add(col_counts, idxs, 1)
+
+    Bp, Aj_mask = _csr_column_index1_indptr(
+        idx_map, sort_idxs, col_counts, Ap, Aj)
+
+    return Bp, Aj_mask, col_counts, sort_idxs
+
+
+_csr_column_index2_ker = core.ElementwiseKernel(
```
Comment (asi1024) [inline cupyx/scipy/sparse/_index.py:97]: This kernel looks to work slowly in some cases for the same reasons as https://github.com/cupy/cupy/pull/3658#discussion_r470135917, but it's OK to leave in this PR.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM!

**Pull request merged**